### PR TITLE
ci(pr): use action actions/labeler for auto pr labeling

### PR DESCRIPTION
**Pull request by nikoksr**
## Description

This change will add a config and a workflow for [actions/labeler](https://github.com/marketplace/actions/labeler). This allows for path specific auto-labeling of pull-requests, which helps with organization and search.

## Motivation and Context

PRs never really get labeled and it's probably not the most important thing to change but it still seems helpful to me, to use the labels we got and make PRs a little "completer" automatically. This not only helps making things more organizd but also allows to use GitHubs PR filtering more efficiently.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] CI 

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by nikoksr** [comment]
@svaloumas your feedback on this would be sufficient and welcome!

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+affects/core:
+  - any: [ '/*.go', '!/*_test.go']
+
+# Add affects/services label to all changes to the services directory
+affects/services:
+  - 'services/*'
+
+# Add affects/tests label when any test file is changed
+affects/tests:
+  - '**/*_test.go'
```
Comment (svaloumas) [inline .github/labeler.yml:11]: Would this be sufficient to also match the test files in the root directory, or we may need another `- /*_test.go` item?

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Add affects/core label to all changes to the core package; exclude changes to test files
+affects/core:
+  - any: [ '/*.go', '!/*_test.go']
+
+# Add affects/services label to all changes to the services directory
+affects/services:
+  - 'services/*'
```
Comment (svaloumas) [inline .github/labeler.yml:7]:
```suggestion
  - 'service/*'
```

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  - '**/*_test.go'
+
+# Add affects/docs label when any markdown file or the docs directory is changed
+affects/docs:
+  - 'docs/**/*'
+  - '/*.md'
+
+# Add affects/ci label when any CI related file is changed
+affects/ci:
+  - '.github/**/*'
```
Comment (svaloumas) [inline .github/labeler.yml:20]: If the syntax `**/` does not match files too, but only directories, maybe we should add another one for the files under `.github/` (Currently [PULL_REQUEST_TEMPLATE.md](https://github.com/nikoksr/notify/blob/main/.github/PULL_REQUEST_TEMPLATE.md) and [config.yml](https://github.com/nikoksr/notify/blob/main/.github/config.yml)).

**Pull request closed without merging**